### PR TITLE
GatherElements: support Gather+GatherElements mode

### PR DESCRIPTION
**Pull request by andrejsokolov**
Support (Gather+GatherElements) optimization for GatherElements layer.
Includes PR: https://github.com/openvinotoolkit/openvino/pull/4140

### CODE REVIEW

**Comment by GlebKazantaev** [comment]
As this PR depends on #4140 which must be merged before this PR I set `DO NOT MERGE` label for now.

**Comment by andrejsokolov** [comment]
@AndrewBakalinIntel, @elatkin, please take a look (the last commit only).

**Comment by andrejsokolov** [comment]
> As this PR depends on #4140 which must be merged before this PR I set `DO NOT MERGE` label for now.

ok, thank you.

File: inference-engine/src/vpu/graph_transformer/src/stages/gather_elements.cpp
Comment (AndrewBakalinIntel) [inline inference-engine/src/vpu/graph_transformer/src/stages/gather_elements.cpp:166]: I don't think you need `parseExpGatherElements`, just check the number of inputs in `parseGatherElements` and handle the difference.

File: inference-engine/src/vpu/graph_transformer/src/stages/gather_elements.cpp
Comment (AndrewBakalinIntel) [inline inference-engine/src/vpu/graph_transformer/src/stages/gather_elements.cpp:133]: I think you don't need `addExpGatherElementsStage`, you may pass the inputs as vector there, and depending on its size, set the attribute to `true` or `false`.

**Review by AndrewBakalinIntel — COMMENTED** [review]
In the last state of my PR I have the functional test disabled, pls enable it in your PR:
https://github.com/openvinotoolkit/openvino/pull/4140/files#diff-96c22829f67b8ca6f51e6b160ac5baeb859cc4c41a52fb0fa8f160d6dcbdf7bf

File: inference-engine/src/vpu/graph_transformer/src/stages/gather_elements.cpp
Comment (andrejsokolov) [reply inference-engine/src/vpu/graph_transformer/src/stages/gather_elements.cpp]: Fixed

File: inference-engine/src/vpu/graph_transformer/src/stages/gather_elements.cpp
Comment (andrejsokolov) [reply inference-engine/src/vpu/graph_transformer/src/stages/gather_elements.cpp]: Fixed

**Comment by andrejsokolov** [comment]
> In the last state of my PR I have the functional test disabled, pls enable it in your PR:
> https://github.com/openvinotoolkit/openvino/pull/4140/files#diff-96c22829f67b8ca6f51e6b160ac5baeb859cc4c41a52fb0fa8f160d6dcbdf7bf

Enabled the tests, but for one DSR test, had to reduce the tensor size (Some problem during the sending tensor to device, just before the layer runs)

**Review by AndrewBakalinIntel — APPROVED** [review]

**Comment by AndrewBakalinIntel** [comment]
PR mentioned above has been merged, please rebase on top of the latest master.
BTW, the PR with firmware update has been recently merged: https://github.com/openvinotoolkit/openvino/pull/4178
You must probably need to rebuild the firmware as well.

**Review by elatkin — APPROVED** [review]

**Comment by Maxim-Doronin** [comment]
@andrejsokolov, we have just merged another fw. Please, resolve conflicts. Sorry for inconvenience

File: inference-engine/tests/functional/plugin/myriad/shared_tests_instances/skip_tests_config.cpp
Comment (Maxim-Doronin) [inline inference-engine/tests/functional/plugin/myriad/shared_tests_instances/skip_tests_config.cpp:37]: @andrejsokolov, why did it changed? `1.1.1` topology hangs. Please, abort the test build and revert this change.

**Review by Maxim-Doronin — CHANGES_REQUESTED** [review]

File: inference-engine/tests/functional/plugin/myriad/shared_tests_instances/skip_tests_config.cpp
Comment (andrejsokolov) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/skip_tests_config.cpp]: Fixed. Thank you!

**Comment by andrejsokolov** [comment]
@Maxim-Doronin, could we merge the PR?

**Review by Maxim-Doronin — APPROVED** [review]

**Pull request merged**
